realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -2709,12 +2712,10 @@ void BaseRealSenseNode::publishPointCloud(rs2::points pc, const rclcpp::Time& t,
     sensor_msgs::PointCloud2Modifier modifier(_msg_pointcloud);
     modifier.setPointCloud2FieldsByString(1, "xyz");
     modifier.resize(pc.size());
-    if (_ordered_pc)
-    {
-        _msg_pointcloud.width = depth_intrin.width;
-        _msg_pointcloud.height = depth_intrin.height;
-        _msg_pointcloud.is_dense = false;
-    }
+
+    _msg_pointcloud.width = depth_intrin.width / _pc_subsample_fct;
+    _msg_pointcloud.height = depth_intrin.height / _pc_subsample_fct;
+    _msg_pointcloud.is_dense = !_ordered_pc;
```

Comment:
you should only populate the `_msg_pointcloud` with `width` and `height` when the pointcloud is meant to be ordered, if it is not orderd it shouldn't have that populated, but with the defaults (which are 1=width and 1=height)

Reply:
So when you change this bear in mind where you use `_msg_pointcloud.width`  because this can fuck the things up

Reply:
You are right. Thanks for pointing this out. I will fix it

Reply:
I implemented a variable to store the pointcloud intrinsics and use that in the service instead of the ROS message one. As for not  defining the fields of `width` and `height` of the message, I need to do this in order to subsample the pointcloud by rows more easily, however before publishing the message the height is set to `1` and the width is set to the amount of points inside the pointcloud given the case it is unordered (see [here](https://github.com/kiwicampus/realsense-ros/blob/feature/subsample-pointcloud/realsense2_camera/src/base_realsense_node.cpp#L2834)).

Reply:
Ok!